src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerElementExtensions.wasm.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+
+namespace Windows.UI.Xaml.Controls;
+public static class MediaPlayerElementExtensions
+{
+	private static Dictionary<nint, bool> _pendingMediaElements = new Dictionary<nint, bool>();
+	public static DependencyProperty AnonymousCORSProperty { get; } =
+	DependencyProperty.RegisterAttached(
+		"AnonymousCORS",
+		typeof(bool),
+		typeof(MediaPlayerElementExtensions),
+		new FrameworkPropertyMetadata(false, OnAnonymousCORSChanged));
+
+	public static void SetAnonymousCORS(MediaPlayerElement d, bool value)
+	{
+		d.SetValue(AnonymousCORSProperty, value);
+	}
+
+	private static void OnAnonymousCORSChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+	{
+		if (d is MediaPlayerElement _mediaPlayer)
+		{
+			var _anonymousCORSEnabled = (bool)e.NewValue;
+			if (_mediaPlayer?.MediaPlayer?.Player is { } player)
+			{
+				//Current MediaElement is initialized, so apply value directly to it
+				player.SetAnonymousCORS(_anonymousCORSEnabled);
+				return;
+			}
+			else
+			{
+				//Add current MediaElement to pending list and wait for Loaded event to be triggered
+				if (!_pendingMediaElements.ContainsKey(_mediaPlayer.HtmlId))
+				{
+					_mediaPlayer.Loaded += MediaElement_Loaded;
+					_pendingMediaElements.Add(_mediaPlayer.HtmlId, _anonymousCORSEnabled);
```

Comment:
Instead of maintaining a dictionary, could we capture the new value in a lambda?

Reply:
@Youssef1313 could you please give me an example of this?

Reply:
@arqueror 

```csharp
using System;

namespace Windows.UI.Xaml.Controls;
public static class MediaPlayerElementExtensions
{
	public static DependencyProperty AnonymousCORSProperty { get; } =
		DependencyProperty.RegisterAttached(
			"AnonymousCORS",
			typeof(bool),
			typeof(MediaPlayerElementExtensions),
			new FrameworkPropertyMetadata(false, OnAnonymousCORSChanged));

	public static void SetAnonymousCORS(MediaPlayerElement d, bool value)
	{
		d.SetValue(AnonymousCORSProperty, value);
	}

	private static void OnAnonymousCORSChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
	{
		if (d is MediaPlayerElement mediaPlayer)
		{
			var anonymousCORSEnabled = (bool)e.NewValue;
			if (mediaPlayer.MediaPlayer?.Player is { } player)
			{
				// Current MediaElement is initialized, so apply value directly to it
				player.SetAnonymousCORS(anonymousCORSEnabled);
			}
			else
			{
				EventHandler handler = null;
				handler = (sender, args) =>
				{
					var mediaPlayer = (MediaPlayerElement)sender;
					if (mediaPlayer.MediaPlayer?.Player is { } player)
					{
						player.SetAnonymousCORS(anonymousCORSEnabled);
					}

					mediaPlayer.Loaded -= handler;
				};

				mediaPlayer.Loaded += handler;
			}
		}
	}
}
```

If I understand correctly, this will have the same functionality you want, but avoids the need for having a dictionary.